examples/08_deduplication.py

```diff
@@ -0,0 +1,163 @@
+
+    data = []
+    for example, n_ex in zip(examples, entries_per_example):
+        len_ex = len(example)
+        # generate a 2D array of chars of size (n_ex, len_ex)
+        str_as_list = np.array([list(example)] * n_ex)
+        # randomly choose which characters are misspelled
+        idxes = np.where(
+            np.random.random(len(example[0]) * n_ex) < prob_mistake_per_letter
+        )[0]
+        # and randomly pick with which character to replace
+        replacements = [
+            string.ascii_lowercase[i]
+            for i in np.random.choice(np.arange(26), len(idxes)).astype(int)
+        ]
+        # introduce spelling mistakes at right examples and right char locations per example
+        str_as_list[idxes // len_ex, idxes % len_ex] = replacements
+        # go back to 1d array of strings
+        data.append(np.ascontiguousarray(str_as_list).view(f"U{len_ex}").ravel())
+    return np.concatenate(data)
+
+
+np.random.seed(123)
+
+# our three medication names
+medications = ["Contrivan", "Genericon", "Zipholan"]
+entries_per_medications = [500, 100, 1500]
+
+# 5% probability of a typo per letter
+prob_mistake_per_letter = 0.05
+
+data = generate_example_data(
+    medications, entries_per_medications, prob_mistake_per_letter
+)
+# we extract the unique medication names in the data & how often they appear
+unique_examples, counts = np.unique(data, return_counts=True)
+# and build a series out of them
+ex_series = pd.Series(counts, index=unique_examples)
+
+
+# ## Visualize the data
+
+
+ex_series.plot.bar(figsize=(20, 10))
+_ = plt.xlabel("Medication name")
+_ = plt.ylabel("Counts")
+
+
+# We can now see clearly the structure of the data: The three original medications are the most common ones, however there are many spelling mistakes and hence many slight variations of the names of the original medications.
+#
+# The idea is to use the fact that the string-distance of each misspelled medication name will be closest to either the correctly or incorrectly spelled orginal medication name - and therefore form clusters.
+
+
+
+from dirty_cat.deduplicate import deduplicate, compute_ngram_distance
+from scipy.spatial.distance import squareform
+
+ngram_distances = compute_ngram_distance(unique_examples)
+square_distances = squareform(ngram_distances)
+
+
+# ## We can visualize the pair-wise distance between all medication names
+#
+# Below I use a heatmap to visualize the pairwise-distance between medication names.
+# A darker color means that two medication names are closer together (are more similar), a lighter color means a larger distance.
+# We can see that we are dealing with three clusters - the original medication names and their misspellings that cluster around them.
+
+import seaborn as sns
+
+fig, axes = plt.subplots(1, 1, figsize=(12, 12))
+sns.heatmap(
+    square_distances, yticklabels=ex_series.index, xticklabels=ex_series.index, ax=axes
+)
+
+
+# ### Clustering to suggest corrections of misspelled names
```

Comment:
Can you use sphinx-gallery markup here to do titles and line separation. A bit of doc here:
https://sphinx-gallery.github.io/stable/syntax.html#embedding-rst
(it's mostly using restructured text in the comments, with the proper separator at the top)